### PR TITLE
fix(ObjectViewer) : large values ellipsis

### PR DESCRIPTION
**Pull request by sgendre**
**What is the problem this PR is trying to solve?**
Large values overflow the screen and do not display ellipsis, meaning that the full value can't be read.
https://jira.talendforge.org/browse/TFD-2259

**What is the chosen solution to this problem?**
Display value on one line only + ellipsis and change style on hover to display full value on multiple lines
Make it work on Chrome, Firefirox, and Safari

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/components.sasslint.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 
 src/ObjectViewer/JSONLike/JSONLike.scss
-  65:21  warning  Hex notation should all be upper case  hex-notation
+  63:21  warning  Hex notation should all be upper case  hex-notation
```
Comment (talend-glorieux) [inline output/components.sasslint.txt]: Could you please fix that errors while you're working on JSONLike.scss?

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 2]
```diff
@@ -49,7 +47,7 @@ export function NativeValue({ data, edit, onSelect, onChange, jsonpath, selected
 			type="button"
 			onClick={e => stopAndSelect(e, { onSelect, jsonpath })}
 		>
-			{display}
+			<div className={theme['inner-button-text']}>{display}</div>
```
Comment (talend-glorieux) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js]: I'd rather use a span than a div.

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.scss [hunk 3]
```diff
@@ -101,15 +99,37 @@
...
 		color: $brand-primary;
-		display: inline;
 		font-size: $object-viewer-value-font-size;
-		vertical-align: baseline;
 		border: none;
 		background: none;
 		padding: 0;
 		cursor: text;
 		margin-left: 4px;
+
```
Comment (talend-glorieux) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.scss]: Why the extra newline?

File: packages/components/stories/ObjectViewer.js [hunk 4]
```diff
@@ -219,6 +221,17 @@ const openedNativeTypeHandler = {
 	onChange: action('onChange'),
 };
 
+const rootOpenedTypeHandler = {
+	edited: [],
+	opened: ['$', '$[0]'],
+	onClick: action('onClick'),
+	onSelect: (e, jsonpath) => {
+		selectedJsonpath = jsonpath;
+	},
```
Comment (talend-glorieux) [inline packages/components/stories/ObjectViewer.js]: No need from brackets.

File: output/components.sasslint.txt (hunk 1 above)
Comment (sgendre) [reply output/components.sasslint.txt]: Prettier is in conflict with linter. We have to choose which one is right.

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js [hunk 5]
```diff
@@ -49,7 +47,7 @@ export function NativeValue({ data, edit, onSelect, onChange, jsonpath, selected
 			type="button"
 			onClick={e => stopAndSelect(e, { onSelect, jsonpath })}
 		>
-			{display}
+			<span className={theme['inner-button-text']}>{display}</span>
```
Comment (jsomsanith) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js]: It seems that this span is not needed.
You added the css to make the ellipsis on the button and this span.
--> remove this extra span please

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.scss [hunk 6]
```diff
@@ -101,15 +99,36 @@
...
+	.line-value:hover {
+		display: inline-block;
+		white-space: initial;
+		word-break: break-all;
+		overflow: visible;
+		vertical-align: middle;
+	}
+
+	.inner-button-text {
```
Comment (jsomsanith) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.scss]: You can remove this class with the span removal

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.scss [hunk 7]
```diff
@@ -101,15 +99,36 @@
...
+		display: contents;
+		overflow: hidden;
+		white-space: nowrap;
+		text-overflow: ellipsis;
+	}
+
+	.inner-button-text:hover {
```
Comment (jsomsanith) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.scss]: You can remove this class with the span removal

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.scss [hunk 8]
```diff
@@ -90,14 +89,26 @@
...
 		cursor: text;
 		margin-left: 4px;
+		display: inline-block;
+		overflow: hidden;
+		white-space: nowrap;
+		text-overflow: ellipsis;
+	}
+
+	.line-value:hover {
+		display: inline-block;
```
Comment (frassinier) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.scss]: already inline-block, right?

File: packages/components/src/ObjectViewer/JSONLike/JSONLike.scss [hunk 9]
```diff
@@ -90,14 +89,25 @@
 	}
 
 	.line-value {
+		max-width: 90%;
```
Comment (acateland) [inline packages/components/src/ObjectViewer/JSONLike/JSONLike.scss]: can this be a right-padding with a fixed size ?

**Pull request merged**